src/main.cpp

```diff
@@ -2128,18 +2128,24 @@ int64_t GetProofOfStakeReward(uint64_t nCoinAge, int64_t nFees, std::string cpid
     {
             // Research Age Subsidy - PROD
             int64_t nBoinc = ComputeResearchAccrual(nTime, cpid, operation, pindexLast, VerifyingBlock, VerificationPhase, dAccrualAge, dMagnitudeUnit, AvgMagnitude);
-            int64_t nInterest = nCoinAge * GetCoinYearReward(nTime) * 33 / (365 * 33 + 8);
+            int64_t nInterest = 0;
 
             // TestNet: For any subsidy < 30 day duration, ensure 100% that we have a start magnitude and an end magnitude, otherwise make subsidy 0 : PASS
             // TestNet: For any subsidy > 30 day duration, ensure 100% that we have a midpoint magnitude in Every Period, otherwise, make subsidy 0 : In Test as of 09-06-2015
             // TestNet: Ensure no magnitudes are out of bounds to ensure we do not generate an insane payment : PASS (Lifetime PPD takes care of this)
             // TestNet: Any subsidy with a duration wider than 6 months should not be paid : PASS
 
             /* Constant Block Reward */
-            AppCacheEntry oCBReward= ReadCache("constblkreward","constblkreward");
-            int64_t nCBReward = RoundFromString(oCBReward.value,12);
-            if(nCBReward && fTestNet && GetBoolArg("-constblkreward",true))
+            if (pindexLast->nVersion>=10)
+            {
+                AppCacheEntry oCBReward= ReadCache("protocol","blockreward1");
```

Comment:
We will need a way to refresh this every 360000 blocks or loading nodes will not have the AppCache entry and use 0 as reward. We can time a full chain scan from 0 to now to see how long it actually takes.

Reply:
IIRC block messages are loaded even during loading. That was one of the changes for new tally and reorganize refactor. I only see as a problem, that the message could expire. It should be asserted that it expires consistently across network (eg: check timestamp of the entry object), which it currently does not. Thanks for spotting.

Reply:
If we are to keep the 6 month limit we will need to add functionality for admin message fetching (instead of going straight to Appcache) so we can control the expiration times. It is a relatively small change and should have been done regardless.

Reply:
I do not understand what the problem is.

Reply:
Messages retain in RAM even when they are expired, so in the case that we do not refresh the values every 6 months then long (>6 months) running wallet will use the old CBR value while loading wallets will use 0 GRC.

So, since we have several messages which are bound by heights we should treat them as a group of messages and add functionality to fetch them with the boundaries. We almost do it with beacons but we fetch them by age instead of height which is a bug IMO. The CBR rate message should fall into the same group.